ARROW-1679: [GLib] Add garrow_record_batch_reader_read_next()

kou opened the pull request:
garrow_record_batch_reader_read_next_record_batch() is deprecated.
It's for following C++ API change.

wesm commented:
@kou can you update the JIRA number to 1679?

wesm commented:
Also the build is broken, I'm not sure why it passed in your Meson PR but failing now

kou commented:
Oops, I've fixed title.

Some tests are failed because GObject Introspection on Ubuntu 14.04 is old. I've added version check. It will fix test failures.

kou commented:
> Also the build is broken, I'm not sure why it passed in your Meson PR but failing now

Because I've released a new gobject-introspection gem yesterday. These failed tests has gobject-introspction gem version check (not GObject Introspection itself). These failed tests are enabled with the new gobject-introspection gem.

wesm commented:
thanks!
